File: 프로그래머스/2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.js
const solution = (str1, str2) => {
    
    str1 = str1.toUpperCase()
    str2 = str2.toUpperCase()
    const arr1 = new Array()
    const arr2 = new Array()
    
    for (let i = 0; i < str1.length - 1; i++) {
        const str = str1.substr(i, 2)
        if (str[0] >= "A" && str[0] <= "Z" && str[1] >= "A" && str[1] <= "Z") {
        arr1.push(str)
        }
    }
    for (let i = 0; i < str2.length - 1; i++) {
        const str = str2.substr(i, 2)
        if (str[0] >= "A" && str[0] <= "Z" && str[1] >= "A" && str[1] <= "Z") {
            arr2.push(str)
        }
    }

    const intersection = []
    const union = []
    for (let i = 0; i < arr2.length; i++) {
        if (arr1.indexOf(arr2[i]) >= 0) {
            intersection.push(arr1.splice(arr1.indexOf(arr2[i]), 1))
        }
        union.push(arr2[i])
    }

    for (let i = 0; i < arr1.length; i++) {
        union.push(arr1[i])
    }
    if (intersection.length === 0 && union.length === 0) {
        return 65536
    }
    return parseInt(65536 * (intersection.length / union.length))
}

File: 프로그래머스/2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.py
def solution(str1, str2):
    str1 = str1.upper()
    str2 = str2.upper()
    arr1 = []
    arr2 = []
    
    for i in range(len(str1) - 1):
        s = str1[i:i+2]
        if s[0].isalpha() and s[1].isalpha():
            arr1.append(s)
            
    for i in range(len(str2) - 1):
        s = str2[i:i+2]
        if s[0].isalpha() and s[1].isalpha():
            arr2.append(s)

    intersection = []
    
    temp_arr1 = list(arr1)
    union_elements = [] 
    
    for item2 in arr2:
        if item2 in temp_arr1:
            intersection.append(item2)
            temp_arr1.remove(item2)
        union_elements.append(item2)

    for item1 in temp_arr1:
        union_elements.append(item1)
    
    len_intersection = len(intersection)
    len_union = len(union_elements)

    if len_intersection == 0 and len_union == 0:
        return 65536
        
    if len_union == 0:
        return 0
    
    return int(65536 * (len_intersection / len_union))